### PR TITLE
Social Previews | Hide Mastodon post preview when no connection

### PR DESCRIPTION
**Pull request by manzoorwanijk**
For Mastodon you see the "Your Post" section even if there is no account connected. This PR fixes that.

## Proposed Changes

* Hide Mastodon post preview when there is no connection
* Display link preview as an anonymous user even when the account is connected.

## Testing Instructions

- Checkout fire up this PR or use the Live link below
- Goto `/marketing/connections/:site`
- Ensure that there is no Mastodon connection
- Goto `/posts/:site` 
- Click on the 3 dots for any post and Click "Share"
- Click on "Preview"
- Select Mastodon preview
- Confirm that you only see the link preview and not the post preview
- Now goto `/marketing/connections/:site` again and connect a Mastodon account
- Reload /posts/:site`
- Click on the 3 dots for any post and Click "Share"
- Click on "Preview"
- Select Mastodon preview
- Confirm that you see both the post and link preview
- Confirm that the link preview is shown as an anonymous user

<table>
	<thead>
		<tr>
			<th>
				BEFORE
			</th>
			<th>
				AFTER
			</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<th colspan="2">
				When NO Mastodon connection exists
			</th>
		</tr>
		<tr>
			<td>
				Screenshot 2023-09-13 at 11 34 15 AM
			</td>
			<td>
				Screenshot 2023-09-13 at 11 33 49 AM
			</td>
		</tr>
		<tr>
			<th colspan="2">
				When Mastodon connection exists
			</th>
		</tr>
		<tr>
			<td>
				Screenshot 2023-09-13 at 11 32 21 AM
			</td>
			<td>
				Screenshot 2023-09-13 at 11 33 03 AM
			</td>
		</tr>
	</tbody>

</table>

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/social-previews/src/mastodon-preview/previews.tsx [hunk 1]
```diff
@@ -6,48 +6,59 @@ import { MastodonLinkPreview } from './link-preview';
 import { MastodonPostPreview } from './post-preview';
 import { MastodonPreviewProps } from './types';
 
-export type MastodonPreviewsProps = MastodonPreviewProps & SocialPreviewsBaseProps;
-
-export const MastodonPreviews: React.FC< MastodonPreviewsProps > = ( props ) => {
-	const { headingLevel, isSocialPost } = props;
+export type MastodonPreviewsProps = MastodonPreviewProps &
+	SocialPreviewsBaseProps & {
+		isSocialPost: boolean;
```
Comment (pablinos) [inline packages/social-previews/src/mastodon-preview/previews.tsx]: Should we add this into the SocialPreviewsBaseProps? It's a flag that we'll need for all the previews.

**Review by pablinos — COMMENTED** [review]
Looks good. I was just wondering about where the `isSocialPost` prop should live

File: packages/social-previews/src/mastodon-preview/previews.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply packages/social-previews/src/mastodon-preview/previews.tsx]: It'll get removed in the next PR anyway.

Comment (pablinos) [reply packages/social-previews/src/mastodon-preview/previews.tsx]: Ok, so we're not going to have this property at all? It seems odd to move it in this PR then.

Comment (manzoorwanijk) [reply packages/social-previews/src/mastodon-preview/previews.tsx]: The decision to remove the prop was not made at the time of this PR. I have restored it. Thanks

**Review by pablinos — APPROVED** [review]
I think we could probably improve the link preview section as that should focus on the actual link preview rather than have essentially a repetition of the whole post, but that's not for this task!

**Comment by manzoorwanijk** [comment]
> I think we could probably improve the link preview section as that should focus on the actual link preview rather than have essentially a repetition of the whole post, but that's not for this task!

Are Mastodon previews different from what the component depicts?

**Pull request merged**